daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/dsom/SDE.scala

```diff
@@ -49,6 +49,26 @@ class SchemaDefinitionError(
 
 }
 
+class SchemaDefinitionErrorFromWarning(
+  sdw: SchemaDefinitionDiagnosticBase,
```

Comment:
Can this sdw have the type of `SchemaDefinitionWaring` instead of `SchemaDefinitionDiagnosticBase`? This should only be used for warnings so seems like that would be on okay limitation? And would that simplify the `extends` at all? For example, we wouldn't need to pass in warnID since that's already available in sdw. Maybe it just becomes something like:
```scala
SchemaDefinitionErrorFromWarning(
  sdw: SchemaDefinitionWarning
) extends SchemaDefinitionWarning(
    sdw.warnId,
    sdw.schemaContext,
    sdw.annotationContext,
    sdw.kind,
    sdw.args
  )
```
Keeps things as simple as possible and makes the warning message the exact same as a normal warning, just with a different mode name to make it clear it's been escalated to an error?

Reply:
I don't think this would work for RuntimeSchemaDefinitionWarning since it extends SchemaDefinitionDiagnosticBase rather than SchemaDefinitionWarning

Reply:
We could probably change it to extend SDW if it makes sense.

Reply:
Done...I also made RSDE an SDE to keep things consistent